framework/src/base/EigenProblem.C

```diff
@@ -30,40 +33,26 @@ InputParameters
 validParams<EigenProblem>()
 {
   InputParameters params = validParams<FEProblemBase>();
-  params.addParam<unsigned int>("n_eigen_pairs", 1, "The number of eigen pairs");
-  params.addParam<unsigned int>("n_basis_vectors", 3, "The dimension of eigen subspaces");
-#if LIBMESH_HAVE_SLEPC
-  params += Moose::SlepcSupport::getSlepcEigenProblemValidParams();
-#endif
   return params;
 }
 
 EigenProblem::EigenProblem(const InputParameters & parameters)
   : FEProblemBase(parameters),
-    _n_eigen_pairs_required(getParam<unsigned int>("n_eigen_pairs")),
+    _n_eigen_pairs_required(1),
     _generalized_eigenvalue_problem(false),
-    _nl_eigen(new NonlinearEigenSystem(*this, "eigen0"))
+    _nl_eigen(libmesh_make_unique<NonlinearEigenSystem>(*this, "eigen0")),
+    _is_residual_initialed(false)
 {
 #if LIBMESH_HAVE_SLEPC
-  _nl = _nl_eigen;
+  _nl = _nl_eigen.get();
```

Comment:
Actually, I need `_nl_eigen` in `Eigenproblem`. But `FEProblemBase` needs `_nl_eigen`.